### PR TITLE
Feature/make json column queryable

### PR DESCRIPTION
**Pull request by le-doude**

### CODE REVIEW

File: lib/rails_admin/adapters/active_record.rb [hunk 1]
```diff
@@ -56,8 +56,8 @@ def associations
       def properties
         columns = model.columns.reject do |c|
           c.type.blank? ||
-          DISABLED_COLUMN_TYPES.include?(c.type.to_sym) ||
-          c.try(:array)
+            DISABLED_COLUMN_TYPES.include?(c.type.to_sym) ||
+            c.try(:array)
```
Comment (Jwata) [inline lib/rails_admin/adapters/active_record.rb:60]: formatting ?

File: lib/rails_admin/adapters/active_record.rb [hunk 2]
```diff
@@ -178,11 +178,78 @@ def range_filter(min, max)
...
+              build_statement_for_string_or_text
+            when :enum then
+              build_statement_for_enum
+            when :belongs_to_association then
+              build_statement_for_belongs_to_association
+          end
+        end
+
+        def build_statement_for_jsonb
+          return unless ar_adapter == 'postgresql'
```
Comment (Jwata) [inline lib/rails_admin/adapters/active_record.rb:200]: Isn't their any other ways to specify adapter ? or please leave comment here.

File: app/assets/javascripts/rails_admin/ra.filter-box.js [hunk 3]
```diff
@@ -99,6 +103,42 @@
           '<input placeholder="-∞" class="additional-fieldset between input-sm form-control" style="display:' + ((field_operator == "between") ? 'inline-block' : 'none') + ';" type="' + field_type + '" name="' + value_name + '[]" value="' + (field_value[2] || '') + '" /> ' +
           '<input placeholder="∞" class="additional-fieldset between input-sm form-control" style="display:' + ((field_operator == "between") ? 'inline-block' : 'none') + ';" type="' + field_type + '" name="' + value_name + '[]" value="' + (field_value[3] || '') + '" />';
           break;
+        case 'json':
+        case 'jsonb':
```
Comment (Jwata) [inline app/assets/javascripts/rails_admin/ra.filter-box.js]: Some filters don't work on `json` field.

File: lib/rails_admin/adapters/active_record.rb [hunk 4]
```diff
@@ -178,11 +178,78 @@ def range_filter(min, max)
 
         def build_statement_for_type
           case @type
-          when :boolean                   then build_statement_for_boolean
-          when :integer, :decimal, :float then build_statement_for_integer_decimal_or_float
-          when :string, :text             then build_statement_for_string_or_text
-          when :enum                      then build_statement_for_enum
-          when :belongs_to_association    then build_statement_for_belongs_to_association
+            when :jsonb, :json then
```
Comment (Jwata) [inline lib/rails_admin/adapters/active_record.rb]: Some filters don't work on `json` field.

File: lib/rails_admin/adapters/active_record.rb (hunk 2 above)
Comment (le-doude) [inline lib/rails_admin/adapters/active_record.rb:200]: A cleaner way would be to extend the current AR adapter into a specialised postgres 9.4+ adapter and write logic to have it selected automatically.
Then with that we unlock all the json/jsonb options for rails_admin UI.

**Comment by Jwata** [comment]
LGTM!
[LGTM](http://lgtm.in/i/IMBjEF9Np)

[:+1:](http://lgtm.in/u/IMBjEF9Np) [:-1:](http://lgtm.in/r/IMBjEF9Np)

**Pull request merged**